frontend/src/app/components/channel/channel.controller.ts

```diff
@@ -150,4 +171,36 @@ export class BaseChannelController {
 
     return onlineUsers;
   }
+
+  onUserDrop(data: IDraggedData, destinationType: string = 'channel') {
+    if (data.channelFromId === this.name) {
+      return;
+    }
+    data.destinationType = (this.name === 'control' && destinationType !== 'disable') ? 'search' :  destinationType;
+    data.channelToId = this.name;
+    this.$rootScope.$broadcast('channel.dragged', data);
+  }
+
+  onDragUserFrom(data: IDraggedData) {
+    /*Just define  method*/
```

Comment:
worth to put this into the body:
    throw new Error("onDragUserFrom() not implemented in the base class!");